src/core/meshcop/dataset.hpp

```diff
@@ -63,7 +63,7 @@ class Dataset
      * @param[in]  aType       The type of the dataset, active or pending.
      *
      */
-    explicit Dataset(Tlv::Type aType);
```

Comment:
I think we should not add `const` and remove `const` from the corresponding `*.cpp` files.

Reply:
Sure, sounds good.